cppapi/server/zmqeventsupplier.cpp

```diff
@@ -63,6 +63,38 @@ ZmqEventSupplier *ZmqEventSupplier::_instance = NULL;
 /*		       															*/
 /************************************************************************/
 
+ZmqEventSupplier::Future::Future():is_ready_(false),mutex(),cond(&mutex)
+{
+
+}
+void ZmqEventSupplier::Future::set()
+{
+    omni_mutex_lock lk(mutex);
+    is_ready_ = true;
+    cond.signal();
+    
+}
+
+void ZmqEventSupplier::Future::wait()
+{
+    if(!is_ready_)
```

Comment:
Although boolean access shall be atomic on most architectures, I suggest to stay safe and protect it with a mutex anyway. Probably it will also generate a warning when we run tango with thread sanitizer.